Update 2024-10-30-community-meeting.mdx (#677)
wasmCloud/wasmCloud.com: community/2024-10-30-community-meeting.mdx

{/* Add summary of topics during the meeting. Each Topic should have an H3 heading. A list is usually sufficient for the recap */}

### DEMO: [Full-stack Rust framework in the Leptos project](https://github.com/leptos-rs/leptos/pull/3063)

- [Enzo Nocera](https://github.com/raskyld) is a Wasm enthusiast. He has experience working for many cloud providers but his work with wasmCloud is more of a personal passion.

> - Enzo has an exciting project which integrates Wasm (Wasmtime right now but later wasmCloud) with the Leptos project. The Rust ecosystem has been evolving in the background and so Enzo is demoing a full-stack framework to create backend and frontend applications with a single Rust codebase.

Reviewer: ```suggestion
- Enzo has an exciting project which integrates Wasm (Wasmtime right now but later wasmCloud) with the [Leptos project](https://www.leptos.dev/). The Rust ecosystem has been evolving in the background and so Enzo is demoing a full-stack framework to create backend and frontend applications with a single Rust codebase.
```

- The aim was to prove we can use the same codebase to build a server side and browser side service based on modern Wasm standards.
- 2 WASI contracts - filesyetem to serve static asets and WASI HTTP to stand up the server.
- Super reative web app that runs entirely with Wasm as a demonstration of how quickly we can deploy web pages, and how we can build in backend integration.

> - This is one of the nicest experiences we've seen in building a front-end app with back-end integration. [Well worth checking out in the repo.

Reviewer: ```suggestion
- This is one of the nicest experiences we've seen in building a front-end app with back-end integration. Well worth checking out in [the repo](https://github.com/raskyld/leptos-wasmcloud).
```

- Still WIP but Enzo will share the details in Slack.
- As always, it's really hard to express the joy of the demo - check out the recording below for Enzo's session, and ensuing community questions.

> ### DEMO: wasmcloud_component HTTP wrappers

Reviewer: ```suggestion
### DEMO: `wasmcloud_component` HTTP wrappers
```

- This is V2 of the components crate which incudes HTTP wrapper.

> - This was spurred by improved support for Wasm 32 and WASI P2 on a stable toolchain in Rust - this means it is a much nicer experience when starting and compiling with or without `wash-build`.

Reviewer: ```suggestion
- This was spurred by improved support for Wasm P1/P2 on a stable toolchain in Rust - this means it is a much nicer experience when starting and compiling with or without `wash build`.
```

Not sure if we meant P1 here...?

- Brooks shows what the experience looks and feels like now the Rust toolchain is a little more mature.
- This has now been published to our documentation as a template.

> - Now, rather than using wit-bindgen, we can use the wasmCloud component crate when we're writing in Rust.

Reviewer: ```suggestion
- Now, rather than using wit-bindgen, we can use the [`wasmcloud-component` crate](https://crates.io/crates/wasmcloud-component) when we're writing in Rust.
```

- Completely interoperable with existing HTTP crate functionality.

> - Check out our new and improved [Quickstart](https://wasmcloud.com/docs/tour/hello-world) where you can find more details and get hands on. You can get a feel for what it feels like to write this kind of code.

Reviewer: ```suggestion
- Check out our [new and improved Quickstart](https://wasmcloud.com/docs/tour/hello-world) where you can find more details and get hands on. You can get a feel for what it feels like to write this kind of code.
```

- It's super tricky to represent the demo in notes - check out the recording below for Enzo's session, and ensuing community questions.

### DEMO and DISCUSSION: wkg: how it works + what's next?

> - `wkg` (pronounced wackage) is a package manager for WIT interface files created as part of the wasm-pkg-tools initiative from the Bytecode Alliance. It is comparable to NPM or Cargo, but for building and running applications with components.

Reviewer: ```suggestion
- [`wkg` (pronounced "wackage")](https://github.com/bytecodealliance/wasm-pkg-tools) is a package manager for WIT interface files created as part of the wasm-pkg-tools initiative from the Bytecode Alliance. It is comparable to NPM or Cargo, but for building and running applications with components.
```

- `wkg` manages WIT dependencies packaged as **OCI artifacts** which means any organization that already has an OCI registry can access, store and call standard WIT interfaces.
- Now available in wash 0.36.0!

> - 'wkg' is now LIVE in wash - very exciting as this means it autonanages your dependencies - declare our dependancies in a World file.

Reviewer: ```suggestion
- `wkg` is now used directly in wash - very exciting as this means it automatically manages your dependencies - declared in a WIT world.
```

> - You just run wit-deps and with run wash build it will automagically fetch the dependencies required - locked insde a wasmcloud.lock file.

Reviewer: ```suggestion
- You just run `wash build` and it will automagically fetch the dependencies required - locked insde a wasmcloud.lock file.
```

- This is a great example of how the Wasm component model lends itself to automation by exposing component dependencies from the start.

> - If you want to convert wit-deps or commit the deps directory in this model and need some assistance, head on over to [Slack](https://wasmcloud.slack.com/).

Reviewer: ```suggestion
- If you want to convert from `wit-deps` or commit the deps directory in this model and need some assistance, head on over to [Slack](https://wasmcloud.slack.com/).
```

- [Resource](https://docs.rs/wash-lib/0.30.0/wash_lib/parser/struct.WasmcloudDotToml.html#structfield.package_config).
- Taylor's demo is short, sweet, and well-worth catching up on. Check out the demo and deeper details in the recording below!

## [Q4 roadmap](https://wasmcloud.com/docs/roadmap/) check-in

- Time flies when you're a wasmCloud maintainer. We have made some significant progress in the roadmap and you can follow progress [here](https://wasmcloud.com/docs/roadmap/).
- Notable issues: we have xloisethe omponet sdk go is already in the wild and we've closed several other issies.
- wasmCloud.toml can exist in alternative directories which adds an additional layer of flexibility and repo cleanliness.
- Open PR: host running mutiple lattices. New contributor: working on tab autocomplete for resources in wasmCloud. Human time values - new contribution - super cool.

> - New Quickstarts for Rust ad

Reviewer: ```suggestion
- New Quickstarts for Rust and Typescript
```
